packages/fxa-auth-server/lib/oauth/logging/index.ts

```diff
@@ -2,28 +2,32 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-let mozlog = null;
+type MozLog = { logger: Logger };
+let mozlog: null | MozLog;
 
-module.exports = function(nameOrLog) {
+module.exports = function(nameOrLog: string | MozLog): Logger | never | void {
```

Comment:
Why is this `never` necessary?

Reply:
It's _really_ necessary, but because the function could potentially throw an exception it has the possibility to `never` return anything.

Reply:
Sure it doesn't hurt anything.  But it also doesn't do anything since `never` is a subtype of every type, in a union it gets absorbed into the other types.

Reply:
Got it! This is precisely what I'm looking for in these TS reviews 😄